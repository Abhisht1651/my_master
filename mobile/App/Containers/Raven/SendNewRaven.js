// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { View, Text, Image, FlatList, TouchableOpacity, Dimensions, Platform } from 'react-native'
import { Images } from '../../Themes'
import styles from '../Styles/RavenStyles/SendNewRavenStyles'
import StoryLine from '../../Components/StoryLineShortesCard'

type Props = {
  item: {
    seen: Number,
    total: Number,
    id: String
  }
} & NavigationContainerProps

const Width = Dimensions.get('window').width
const Data = [
  {
    id: '1',
    seen: 50,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '2',
    seen: 95,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '3',
    seen: 155,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '4',
    seen: 500,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '5',
    seen: 350,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '6',
    seen: 45,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '7',
    seen: 415,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  },
  {
    id: '8',
    seen: 450,
    total: 500,
    category: 'World',
    storyTitle: 'UK exit from the European Union',
    date: 'Sept 21, 2019',
    description: 'Ray Contreras talks about the different accents',
    source: 'Forbes',
    trending: true,
    storyImage: Images.storyImage
  }
]
const SendNewRaven = (props: Props) => {
  return (
    <View style={styles.container}>
      {Platform.OS === 'ios' && <View style={styles.topcover} />}
      <View style={styles.container1}>
        <View style={styles.backbutton}>
          <TouchableOpacity onPress={() => props.navigation.goBack()}>
            <Image source={Images.arrowleft} style={styles.arrowleft} />
          </TouchableOpacity>
        </View>
        <View style={styles.text}>
          <Text style={styles.sendraven}>Send a Raven</Text>
        </View>
      </View>
      <View style={styles.container2}>
        <FlatList
          data={Data}
          renderItem={({ item }) => (
            <View style={styles.flatlistcont}>
              <View style={styles.card}>
                <TouchableOpacity onPress={() => props.navigation.navigate('SendRaven', { item })}>
                  <StoryLine
                    type={1}
                    storyTitle={item.storyTitle}
                    date={item.date}
                    description={item.description}
                    source={item.source}
                    storyImage={item.storyImage}
                    category={item.category}
                    trending={item.trending}
                  />
                </TouchableOpacity>
              </View>
              <View style={styles.border} />
              <View style={styles.status}>
                <View style={styles.readstoryfriend}>
                  <View style={styles.outerbar} />
                  <View
                    style={[
                      styles.innerbar,
                      { width: (item.seen / item.total) * (296 / 375) * Width }
                    ]}
                  />
                </View>
                <View style={styles.readstoryfriendcont}>
                  <TouchableOpacity onPress={() => props.navigation.navigate('FriendReadStory')}>
                    <View>
                      <View>
                        <Text style={styles.readfriendtext}>
                          {item.seen}/{item.total} have read the storyline
                        </Text>
                      </View>
                      <View style={styles.chevron}>
                        <Image source={Images.right} style={styles.right} />
                      </View>
                    </View>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
          )}
          keyExtractor={(item) => item.id}
          style={styles.flatlist}
        />
      </View>
    </View>
  )
}

export default SendNewRaven
