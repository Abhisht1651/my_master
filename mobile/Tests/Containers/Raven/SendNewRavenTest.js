import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, FlatList, Text } from 'react-native'
import SendNewRaven from '../../../App/Containers/Raven/SendNewRaven'
// Test for RavenAll (snapshot and props)
describe('SendNewRaven', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<SendNewRaven />) // no compile-time error
  })
  it('RavenAll component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('RavenAll BottomNav', () => {
    expect(wrapper.find('BottomNav')).toHaveLength(0)
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 FlatList', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
  it('should render 1 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(1)
  })
})
