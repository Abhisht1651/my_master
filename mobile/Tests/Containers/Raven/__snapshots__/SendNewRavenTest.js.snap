// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendNewRaven RavenAll component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.623046875,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 1.7369791666666665,
        "borderColor": "#E6ECED",
        "flex": 62,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "left": 53.99999999999999,
          "position": "absolute",
          "top": 25.186197916666664,
          "width": 44,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "width": 44,
            }
          }
        />
      </ForwardRef>
    </View>
    <View
      style={
        Object {
          "width": 400,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 40,
            "left": 251,
            "letterSpacing": -0.22,
            "lineHeight": 53.99999999999999,
            "position": "absolute",
            "textAlign": "center",
            "top": 16,
          }
        }
      >
        Send a Raven
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F7F7F7",
        "flex": 706,
      }
    }
  >
    <FlatList
      data={
        Array [
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "1",
            "seen": 50,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "2",
            "seen": 95,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "3",
            "seen": 155,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "4",
            "seen": 500,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "5",
            "seen": 350,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "6",
            "seen": 45,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "7",
            "seen": 415,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
          Object {
            "category": "World",
            "date": "Sept 21, 2019",
            "description": "Ray Contreras talks about the different accents",
            "id": "8",
            "seen": 450,
            "source": "Forbes",
            "storyImage": Object {
              "default": "",
            },
            "storyTitle": "UK exit from the European Union",
            "total": 500,
            "trending": true,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</View>
`;
